Cython/Compiler/CythonScope.py

```diff
@@ -127,6 +127,17 @@ def load_cythonscope(self):
                                             self.viewscope, cython_scope=self,
                                             whitelist=MemoryView.view_utility_whitelist)
 
+        # Marks the types as imported so that they can be extended from
+        # without Cython attempting to import cython.view
+        ext_types = [ entry.type
+                         for entry in view_utility_scope.entries.values()
+                         if entry.type.is_extension_type ]
+        for mod in self.context.modules.values():
+            if mod is self or mod.is_builtin_scope:
+                continue
+            mod.types_imported.update(ext_types)
```

Comment:
Hmm, this doesn't quite feel right at first sight. Why should we add the internal types to the imported types of all user defined modules? Shouldn't that happen only (and automatically) when a user cimports `cython.view`?

Reply:
I've updated it with a different approach. I can't really work out where it'd be sensible detect that these types are being used and add them to `types_imported` so I've marked them with a flag so they don't get imported instead..